src/app/activity/bounty/Publish.ts

```diff
@@ -1,35 +1,45 @@
+import { TextChannel, Message } from 'discord.js'
+import Log from '../../utils/Log';
 import mongo, { Db, UpdateWriteOpResult } from 'mongodb';
 import MongoDbUtils from '../../utils/MongoDbUtils';
 import { BountyCollection } from '../../types/bounty/BountyCollection';
 import { CustomerCollection } from '../../types/bounty/CustomerCollection';
 import DiscordUtils from '../../utils/DiscordUtils';
 import BountyUtils from '../../utils/BountyUtils';
-import RuntimeError from '../../errors/RuntimeError';
 import { PublishRequest } from '../../requests/PublishRequest';
 import { BountyStatus } from '../../constants/bountyStatus';
 import { Activities } from '../../constants/activities';
 import { Clients } from '../../constants/clients';
+import Bounty from '../../commands/bounty/Bounty';
 
 export const publishBounty = async (publishRequest: PublishRequest): Promise<any> => {
 	Log.debug('In Publish activity');
 
-	Log.info(`starting to finalize bounty: ${publishRequest.bountyId} from guild: ${publishRequest.guildId}`);
 	const bountyId = publishRequest.bountyId;
 	const guildId = publishRequest.guildId;
+
+	Log.info(`starting to finalize bounty: ${publishRequest.bountyId} from guild: ${publishRequest.guildId}`);
+
 	const { guildMember } = await DiscordUtils.getGuildAndMember(publishRequest.guildId, publishRequest.userId);
 
 	const [dbBountyResult, dbCustomerResult] = await getDbHandler(bountyId, guildId, publishRequest);
 
-	if (!publishRequest.clientSyncRequest) {
+	if (!publishRequest.clientSyncRequest && dbBountyResult.status == BountyStatus.draft) {
 		await writeDbHandler(dbBountyResult);
 	}
 
-	const bountyChannel: TextChannel = dbBountyResult.createdInChannel ? await DiscordUtils.getTextChannelfromChannelId(dbBountyResult.createdInChannel) :
+	let bountyChannel: TextChannel;
+	// Possible channel was deleted prior to publish or refresh
+	try {
+	    bountyChannel = dbBountyResult.createdInChannel ? await DiscordUtils.getTextChannelfromChannelId(dbBountyResult.createdInChannel) :
 		await guildMember.client.channels.fetch(dbCustomerResult.bountyChannel) as TextChannel;
+	} catch(e) {
+		bountyChannel = undefined;
+	}
 	const bountyMessage: Message = await BountyUtils.canonicalCard(dbBountyResult._id, publishRequest.activity, bountyChannel, guildMember);
-	await DiscordUtils.activityResponse(publishRequest.commandContext, publishRequest.buttonInteraction,  `Bounty published to \`${(bountyMessage.channel as any).name || bountyChannel.name}\` <${bountyMessage.url}> and the website! <${process.env.BOUNTY_BOARD_URL}${bountyId}>`);
+	if (dbBountyResult.status == BountyStatus.draft || publishRequest.commandContext) {
+		await DiscordUtils.activityResponse(publishRequest.commandContext, publishRequest.buttonInteraction,  `Bounty published to \`${(bountyMessage.channel as any).name || bountyChannel.name}\` <${bountyMessage.url}> and the website! <${process.env.BOUNTY_BOARD_URL}${bountyId}>`);
+	}
```

Comment:
I can't comment directly on the line, but this comment is directed towards line 47

Why do we not need to gate this code block by whether the publish activity was conducted for a draft bounty or for a non-draft bounty? 

```
	if (dbBountyResult.creatorMessage !== undefined) {
		const dmChannel = await guildMember.client.channels.fetch(dbBountyResult.creatorMessage.channelId) as TextChannel;
		const previewMessage = await dmChannel.messages.fetch(dbBountyResult.creatorMessage.messageId);
		await previewMessage.delete();
	}
```

Reply:
That is old code left over from when we stored the DM ID in the creatorMessage. It was left there in case there were any old bounty DMs still around from before the canonicalCard change.